### PR TITLE
Improve HDR rendering

### PR DESCRIPTION
**Pull request by guycalledfrank**
If you tried rendering the same mesh instance to a HDR and non-HDR target before, it would force regeneration of all scene shaders, because tonemap/gamma had to change. Luckily it didn't actually recompile the shaders, but still would perform the full shader key construction for every scene shader and look it up in the cache. It was quite slow.

Instead we now separate meshInstance shader variants to SHADER_FORWARD and SHADER_FORWARDHDR, with latter not using any tonemapping or output gamma correction. As they don't overwrite each other, every object can be safely rendered into HDR/LDR targets when needed.

Also added WebGL 2.0 HDR formats when checking if you're "rendering to HDR".

* can be possibly implemented better

### CODE REVIEW

File: src/graphics/program-lib/program-lib.js [hunk 1]
```diff
@@ -1,19 +1,28 @@
...
-        return value===pc.GAMMA_NONE? pc.shaderChunks.gamma1_0PS :
-              (value===pc.GAMMA_SRGBFAST? pc.shaderChunks.gamma2_2FastPS : pc.shaderChunks.gamma2_2PS);
+
+        if (value === pc.GAMMA_SRGB) {
+            return pc.shaderChunks.gamma2_2PS;
+        } else if (value === pc.GAMMA_SRGBFAST) {
+            return pc.shaderChunks.gamma2_2FastPS;
+        } else if (value === pc.GAMMA_SRGBHDR) {
+            return "#define HDR\n" + pc.shaderChunks.gamma2_2PS;
+        } else if (value === pc.GAMMA_SRGBFASTHDR) {
```
Comment (Maksims) [inline src/graphics/program-lib/program-lib.js]: We've spoke to remove `pc.GAMMA_SRGBFAST` as it is very tiny benefit for only old hardware.
Does it worth adding `pc.GAMMA_SRGBFASTHDR` as well?

Comment (guycalledfrank) [inline src/graphics/program-lib/program-lib.js]: Well, it's still a benefit that made some difference in Seemore on the Mali tablet.

Comment (Maksims) [inline src/graphics/program-lib/program-lib.js]: Are we talking big difference here? This seems like very-very-very tiny difference would make for very-very narrow use case, but creates some branching and confusing options for developers.

Comment (guycalledfrank) [inline src/graphics/program-lib/program-lib.js]: I don't remember, but it wasn't absolutely tiny. The tablet we tested on, is still not too old, and there are a lot of worse cheaper devices still, so I'm not convinced we should remove it.

Comment (Maksims) [inline src/graphics/program-lib/program-lib.js]: Can you provide some tests to prove performance difference?

**Comment by guycalledfrank** [comment]
Alright, gamma2.2_fast is removed

**Pull request merged**